pkg/options/rawyamltmpl/raw_text_applier.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package rawtexttmpl is a special-case component maker for constructing
+// Package rawtexttmpl is a special-case option applier for constructing
 // objects with the assumption that objects have been inlined via RawText into
 // ConfigMaps.
```

Comment:
Why doesn't `SimpleApplier` work here?  I'd expect it to handle ConfigMaps effectively.

Reply:
The raw text converter filters based on the inline annotation and then returns the sub-objects (rather just returning the reifed config maps).

Reply:
Ooohhhh that clarifies this substantially.  Your comment above,

> The returned component should be copy of the original, with (perhaps) modifications made to the original.

seems to directly contradict this, though.  I think these concerns ought to be separated (ConfigMap containing templated objects -> ConfigMap containing complete objects -> complete objects).

Reply:
Added some comments. Does this help?

Reply:
Way better, yes, thanks.